### PR TITLE
Us 25

### LINKED ISSUE #25: User Story 25, Visitors must register or log in to check out
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page

### LINKED ISSUE #26: User Story 26, Registered users can check out
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out and fill out order info and click create order
An order is created in the system, which has a status of "pending"
That order is associated with my user
I am taken to my orders page ("/profile/orders")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty

### PR DESCRIPTION
**Pull request by AngelaGuardia**
closes #25 
closes #26 

**Schema Changes:**
- Changed order status to integer so it takes the enum values

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.bigint "user_id"
-    t.string "status"
+    t.integer "status", default: 0
```
Comment (lkriffell) [inline db/schema.rb:61]: Nice change from a string to an integer and adding the default value

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -46,7 +46,13 @@
 </table>
   <p>Total: <%= number_to_currency(cart.total) %></p>
   <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
-  <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+  <section id="bottom-text">
+    <% if current_user %>
+      <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+    <% else %>
+      <p>You must <%= link_to "register", "/register" %> or <%= link_to "log in", "/login" %> to finish the checkout process</p>
```
Comment (lkriffell) [inline app/views/cart/show.html.erb:53]: I like how it gives you the links within the message to either register or login

**Review by lkriffell — APPROVED** [review]
Looks good. I saw you tested both the happy and sad paths

File: spec/features/cart/show_spec.rb [hunk 3]
```diff
@@ -55,8 +55,51 @@
...
+        it "I must register or login before checking out" do
+          visit '/cart'
+
+          expect(page).to_not have_link("Checkout")
+
+          within("#bottom-text") do
+            expect(page).to have_content("You must register or log in to finish the checkout process")
+            expect(page).to have_link("register")
+            expect(page).to have_link("log in")
+          end
```
Comment (KieraAllen) [inline spec/features/cart/show_spec.rb:69]: This is an awesome check!

File: spec/features/items/index_spec.rb [hunk 4]
```diff
@@ -63,8 +63,10 @@
 
       flying_disc = @brian.items.create(name: "A Flying Disc", description: "Flies for days!", price: 10, image: "https://hw.menardc.com/main/items/media/CEGEN001/ProductLarge/253-0107_P_4.jpg", inventory: 32)
 
-      order = Order.create!(name: "name", address: "address", city: "city", state: "state", zip: 80210)
-      order_2 = Order.create!(name: "name", address: "address", city: "city", state: "state", zip: 80210)
+      user = create(:user)
```
Comment (KieraAllen) [inline spec/features/items/index_spec.rb:66]: This is created using FactoryBot, right? Love it.

**Review by KieraAllen — APPROVED** [review]

File: spec/features/items/index_spec.rb (hunk 4 above)
Comment (AngelaGuardia) [reply spec/features/items/index_spec.rb:66]: yup!

File: app/models/order.rb [hunk 5]
```diff
@@ -5,6 +5,8 @@ class Order < ApplicationRecord
   has_many :items, through: :item_orders
   belongs_to :user
 
+  enum status: %w(pending packaged shipped cancelled)
```
Comment (Arique1104) [inline app/models/order.rb:8]: This is so great and would've been so useful for me and the stories I'm working on.

File: db/migrate/20200915231551_change_order_status.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class ChangeOrderStatus < ActiveRecord::Migration[5.2]
+  def change
+    change_column :orders, :status, 'integer USING CAST(status AS integer)'
```
Comment (Arique1104) [inline db/migrate/20200915231551_change_order_status.rb:3]: I wondered why I had access to a :status column.

**Review by Arique1104 — APPROVED** [review]
Nice work!

**Pull request merged**